waterlog/src/Components/DailyCostsReports.js

```diff
@@ -1,9 +1,19 @@
 import React from 'react';
 import { Line, defaults } from 'react-chartjs-2';
 
-const DailyCostsReports = (props) => { 
-  let labelX = props.props.dataPoints.map(a => (new Date(a.x).getHours() + ":00"));
-  let dataY = props.props.dataPoints.map(a => Math.round(a.y)); 
+const DailyCostsReports = (props) => {
+  let labelX = props.props.dailyCost.dataPoints.map(a => (new Date(a.x).getHours() + ":00"));
+  let dataY = props.props.dailyCost.dataPoints.map(a => Math.round(a.y)); 
+  const {yIntercept, slope} = props.props.forecastDaily[0];
+
+  const startX = Math.floor((new Date(props.props.dailyCost.dataPoints[0].x).getTime()) / 1000); 
```

Comment:
Please extract the `new Date(props.props.dailyCost.dataPoints[0].x).getTime()` and `new Date(props.props.dailyCost.dataPoints[labelX.length-1].x).getTime()` into variables

Reply:
Moved into variables.